src/commands/generate.ts

```diff
@@ -0,0 +1,119 @@
+import { flags } from '@oclif/command';
+import Command from '../base';
+import Generator from '@asyncapi/generator';
+import path from 'path';
+import os from 'os';
+import { load } from '../models/SpecificationFile';
+
+export default class Generate extends Command {
+    static description = 'Generator is a tool that you can use to generate whatever you want basing on the AsyncAPI specification file as an input.';
+
+    static examples: string[] | undefined = [
+      'asyncapi generate asyncapi.yaml @asyncapi/html-template'
+    ]
+
+    static flags = {
+      help: flags.help({ char: 'h' }),
+      'disable-hook': flags.string({
+        char: 'd',
+        description: 'disable a specific hook type or hooks from a given hook type'
+      }),
+      install: flags.boolean({
+        char: 'i',
+        default: false,
+        description: 'installs the template and its dependencies (defaults to false)'
+      }),
+      debug: flags.boolean({
+        description: 'to enable specific errors in the console'
+      }),
+      'no-overwrite': flags.string({
+        char: 'n',
+        description: 'glob or path of the file(s) to skip when regenerating'
+      }),
+      output: flags.string({
+        char: 'o',
+        description: 'directory where to put the generated files (defaults to current directory)'
+      }),
+      'force-write': flags.boolean({
+        default: false,
+        description: 'force writing of the generated files to given directory even if it is a git repo with unstaged files or not empty dir (defaults to false)'
+      }),
+      'watch-tempalte': flags.boolean({
```

Comment:
@Souvikns Some tip: you can use https://github.com/asyncapi/cli/blob/master/src/flags.ts#L3 that flag and that https://github.com/asyncapi/cli/blob/master/src/globals.ts#L28 helper to enable watcher :)